src/commands/stepfunctions/awsCommands.ts

```diff
@@ -0,0 +1,328 @@
+  const data = await stepFunctionsClient.send(command)
+
+  return data
+}
+
+export const putSubscriptionFilter = async (
+  cloudWatchLogsClient: CloudWatchLogsClient,
+  forwarderArn: string,
+  filterName: string,
+  logGroupName: string,
+  stepFunctionArn: string,
+  context: BaseContext,
+  dryRun: boolean
+): Promise<PutSubscriptionFilterCommandOutput | undefined> => {
+  // Running this function multiple times would not create duplicate filters (old filter with the same name would be overwritten).
+  // However, two filters with the same destination forwarder can exist when the filter names are different.
+
+  const input = {
+    destinationArn: forwarderArn,
+    filterName,
+    filterPattern: '',
+    logGroupName,
+  }
+  const command = new PutSubscriptionFilterCommand(input)
+  const commandName = 'PutSubscriptionFilter'
+  displayChanges(stepFunctionArn, context, commandName, dryRun, input)
+  if (!dryRun) {
+    const data = await cloudWatchLogsClient.send(command)
+    // Even if the same filter name is created before, the response is still 200.
+    // there are no way to tell
+    context.stdout.write(
+      `Subscription filter ${filterName} is created or the original filter ${filterName} is overwritten.\nt`
+    )
+
+    return data
+  }
+}
+
+export const tagResource = async (
+  stepFunctionsClient: SFNClient,
+  stepFunctionArn: string,
+  tags: Tag[],
+  context: BaseContext,
+  dryRun: boolean
+): Promise<TagResourceCommandOutput | undefined> => {
+  const input = {
+    resourceArn: stepFunctionArn,
+    tags,
+  }
+
+  const command = new TagResourceCommand(input)
+  const commandName = 'TagResource'
+  displayChanges(stepFunctionArn, context, commandName, dryRun, input)
+  if (!dryRun) {
+    const data = await stepFunctionsClient.send(command)
+    printSuccessfulMessage(commandName, context)
+
+    return data
+  }
+}
+
+export const createLogGroup = async (
+  cloudWatchLogsClient: CloudWatchLogsClient,
+  logGroupName: string,
+  stepFunctionArn: string,
+  context: BaseContext,
+  dryRun: boolean
+): Promise<CreateLogGroupCommandOutput | undefined> => {
+  const input = {
+    logGroupName,
+  }
+  const command = new CreateLogGroupCommand(input)
+  const commandName = 'CreateLogGroup'
+  displayChanges(stepFunctionArn, context, commandName, dryRun, input)
+  try {
+    if (!dryRun) {
+      const data = await cloudWatchLogsClient.send(command)
+      printSuccessfulMessage(commandName, context)
+
+      return data
+    }
+  } catch (err) {
+    // if a resource already exists it's a warning since we can use that resource instead of creating it
+    if (err instanceof Error) {
+      if (err.name === 'ResourceAlreadyExistsException') {
+        context.stdout.write(
+          ` -> [Info] ${err.message}. Skipping resource creation and continuing with instrumentation.\n`
+        )
+      }
+    } else {
+      context.stdout.write(` -> [Error] ${err.message}`)
+    }
```

Comment:
This will never be reached, checking for `instanceof` is just for guarding. It is safe to remove.
```suggestion
    }
```